madara/crates/client/cairo_native/src/cache.rs

```diff
@@ -0,0 +1,1614 @@
+        assert_eq!(NATIVE_CACHE.len(), 0);
+    }
+
+    #[test]
+    fn test_cache_size_metric() {
+        let _guard = TEST_MUTEX.lock().unwrap();
+        // Clear cache
+        NATIVE_CACHE.clear();
+
+        let config = config::NativeConfig::default();
+        // Cache operations should not panic
+        evict_cache_if_needed(&config);
+    }
+
+    #[test]
+    fn test_cache_eviction_logic() {
+        let _guard = TEST_MUTEX.lock().unwrap();
+        // Clear cache
+        NATIVE_CACHE.clear();
+
+        let config = config::NativeConfig::default();
+        let max_size = config.max_memory_cache_size;
+
+        // Test that eviction threshold is calculated correctly
+        // If Some, should be positive; if None, unlimited cache is valid
+        if let Some(size) = max_size {
+            assert!(size > 0, "Max cache size should be positive when set");
+        }
+        // None is also valid (unlimited cache)
+    }
+
+    // ============================================================================
+    // Priority 1.1: Memory Cache Hit Scenarios
+    // ============================================================================
+
+    #[test]
+    fn test_basic_memory_cache_hit() {
+        // Acquire metrics mutex to prevent interference with other tests
+        use crate::metrics::test_counters;
+        let _metrics_guard = test_counters::acquire_and_reset();
+
+        // Use unique class_hash to avoid interference with other parallel tests
+        let class_hash = create_unique_test_class_hash();
+
+        // Create test Sierra class
+        let sierra = create_test_sierra_class().expect("Failed to create test Sierra class");
+        let temp_dir = TempDir::new().expect("Failed to create temp dir");
+
+        // Create native class and insert into cache
+        let native_class =
+            create_test_native_class(&sierra, &temp_dir, class_hash, None).expect("Failed to create test native class");
+
+        // Insert into cache with initial timestamp
+        let initial_time = Instant::now();
+        NATIVE_CACHE.insert(class_hash, (native_class.clone(), initial_time));
+
+        // Verify cache contains our specific class
+        assert!(NATIVE_CACHE.contains_key(&class_hash), "Cache should contain our test class");
+
+        // Try to get from memory cache - should succeed
+        let result = try_get_from_memory_cache(&class_hash, &create_simple_test_config(&temp_dir));
+        assert!(result.is_some(), "Memory cache should return Some when class is cached");
+
+        // Verify cache still contains the class (and access time was updated)
+        assert!(NATIVE_CACHE.contains_key(&class_hash), "Class should still be in cache after access");
+        if let Some(entry) = NATIVE_CACHE.get(&class_hash) {
+            let (_, access_time) = entry.value();
+            assert!(*access_time >= initial_time, "Access time should be updated to current or later time");
```

Comment:
```suggestion
            assert!(*access_time > initial_time, "Access time should be updated to current or later time");
```

`>=` seems like an incorrect check for an update